aten/src/ATen/native/cpu/DistributionKernels.cpp

```diff
@@ -119,10 +119,7 @@ void exponential_kernel(TensorIteratorBase &iter, double lambda, std::optional<G
     {
       // See Note [Acquire lock when using random generators]
       std::lock_guard<std::mutex> lock(generator->mutex_);
-      if (self.scalar_type() == at::kDouble)
-        seed = generator->random64();
-      else
-        seed = generator->random();
+      seed = generator->random();
```

Comment:
Is it fine to always use 32-bit seed?

Reply:
The torch generator generates a 32-bit random number each time, which we use as the seed for `vslNewStream`. Is it enough? Tt may depend on the sequence length torch generator generated before generating the same random number. If a random number is not enough as a seed, I think we can use `vslNewStreamEx` and use two 32-bit random numbers as seeds (same as generating a 64 bit random number as a seed).